core/src/main/java/org/infinispan/interceptors/xsite/BaseBackupInterceptor.java

```diff
@@ -137,15 +137,16 @@ private Object handleSingleKeyWriteReturn(InvocationContext ctx, DataWriteComman
          return rv;
       }
       int segment = dataWriteCommand.getSegment();
-      DistributionInfo dInfo = clusteringDependentLogic.getCacheTopology().getSegmentDistribution(segment);
-      if (dInfo.isWriteOwner()) {
-         //all owners has to keep track of the updates even if the primary owner is the only one who sends it.
+      if (clusteringDependentLogic.getCacheTopology().getSegmentDistribution(segment).isPrimary()) {
+         //primary owner sends for (a)sync backups
```

Comment:
picking the example above ^^, the primary owner sends the request to NYC and SFO.

For IRAC, the backup owner needs to keep track of the keys changed in case the primary owner crashes.

Reply:
> picking the example above ^^, the primary owner sends the request to NYC and SFO.

But why does the comment mention backups at all?

> For IRAC, the backup owner needs to keep track of the keys changed in case the primary owner crashes.

I understood that. What's confusing me is that backups "keep track in remote context", while the primary "sends for (a)sync backups", when they are calling the same method with the same arguments.

Reply:
> But why does the comment mention backups at all?

backups == remote site in this comment. I'll change to avoid confusion

> What's confusing me is that backups "keep track in remote context", while the primary "sends for (a)sync backups", when they are calling the same method with the same arguments.

the primary owner tracks and sends the update for ASYNC remote sites while the backup owner only tracks... I'll try to rephrase the comment.